### ISSUE TITLE
maybe simplification

### ISSUE THREAD

**Issue by colinator** [fzenke/spytorch#3]
I don't understand why the 'rst' variable exists. It seems to always be == 'out'. Changing to `rst = out` yields same results...

```
def spike_fn(x):
    out = torch.zeros_like(x)
    out[x > 0] = 1.0
    return out
...
# Here we loop over time
for t in range(nb_steps):
    mthr = mem-1.0
    out = spike_fn(mthr) 
    rst = torch.zeros_like(mem)
    c = (mthr > 0)
    rst[c] = torch.ones_like(mem)[c] 
```

**Comment by fzenke** [comment]
Thanks for asking this question. The two variables are only the same in the forward pass. When a surrogate gradient is introduced the backward passes are different. Specifically, no gradients flow through the rst variable. In some cases this makes learning more robust. See https://www.biorxiv.org/content/10.1101/2020.06.29.176925v1 Fig 5 for more details on including the reset term in gradient computation. However, we will make this more explicit by simply writing it as rst = out.detach() in a next version.
